tests/functional/flakes/absolute-attr-paths.sh

```diff
@@ -14,6 +14,6 @@ cat > $flake1Dir/flake.nix <<EOF
 }
 EOF
 
-[ "$(nix eval --impure --json $flake1Dir#.x)" -eq 1 ]
-[ "$(nix eval --impure --json $flake1Dir#x)" -eq 2 ]
-[ "$(nix eval --impure --json $flake1Dir#.packages.$system.x)" -eq 2 ]
+[ "$(nix eval --impure --json "$flake1Dir"#.x)" -eq 1 ]
+[ "$(nix eval --impure --json "$flake1Dir"#x)" -eq 2 ]
+[ "$(nix eval --impure --json "$flake1Dir"#.packages."$system".x)" -eq 2 ]
```

Comment:
how did that every work with the `#`!

Reply:
(oh, @roberth says because no whitespace around the `#`.)

Reply:
Speaking of which, I would love to combine things like this into one big `"...."`, but I am fine saving that for a separate PR if it's harder to automate (thank `shellcheck --fix` or whatever may have assisted with this).

Reply:
Yes, the hashtag is only a comment if there is a space before.

I don't think there is an autofixer in shellcheck itself.

Also I think comment rendering could be fixed by quoting the entire string, but that's neglible. 

```suggestion
[ "$(nix eval --impure --json "$flake1Dir#.x")" -eq 1 ]
[ "$(nix eval --impure --json "$flake1Dir#x")" -eq 2 ]
[ "$(nix eval --impure --json "$flake1Dir#.packages.$system.x")" -eq 2 ]
```

Reply:
Most of the commits were the `shellcheck -p diff | git apply`. The only exceptions were adding a shebang to the top of a file or disabling a lint to ignore a false positive